SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPossession/TestPossessionPlayerController.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "TestPossessionPlayerController.h"
+#include "Engine/World.h"
+#include "EngineClasses/Components/RemotePossessionComponent.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Utils/SpatialStatics.h"
+
+DEFINE_LOG_CATEGORY(LogTestPossessionPlayerController);
+
+int32 ATestPossessionPlayerController::OnPossessCalled = 0;
+
+ATestPossessionPlayerController::ATestPossessionPlayerController() {}
+
+void ATestPossessionPlayerController::OnPossess(APawn* InPawn)
+{
+	Super::OnPossess(InPawn);
```

Comment:
If the logic of when this is called was broken, so this were called on an non-auth Pawn, I'm not sure the test would break. Perhaps only increment the OnPoessessCalled var if your auth over both this and InPawn

Reply:
The function OnPossess was called only when Controller and Pawn authoritative in the same worker. 
I think it was not necessary to have an Authority check for Controller and Pawn here.

Reply:
> The function OnPossess was called only when Controller and Pawn authoritative in the same worker.

Where is that guarantee?

Reply:
At the AController::Possess(APawn* InPawn) I see if the Controller doesn't HasAuthority, it will be returned before OnPossess, and also return if InPawn doesn't HasAuthority.
But it has some additional checks. But for our functional tests, it only about HasAuthority or not.
After all, I add the HasAuthority check before increment the OnPoessessCalled and log if not.